url4192078a2f_unsegnor_hipotetizador_tgz-pJ8-hipotetizador_MedirTiemposJ8/src/hipotetizador/Hipo.java

```diff
@@ -0,0 +1,1790 @@
+                        //Si es cero la probabilidad se invierte
+                        float probabilidad = (float) (1 - (float) valorNuevo) - (float) r.getConfianza();
+                        probabilidades[e.getSubindice()][e.getEntrada()] = probabilidad;
+                    }
+                }
+            }
+
+        }
+
+        return respuesta;
+    }
+
+    private ArrayList<Regla> extraer_casos_ambiguos(ArrayList<Regla> reglas) {
+        //Inicializar respuesta
+        ArrayList<Regla> respuesta = new ArrayList<>();
+
+
+        //Ordenar la lista por el antecedente y consecuente
+        ComparaReglasPorAntecedente comp = new ComparaReglasPorAntecedente();
+        Collections.sort(reglas, comp);
+
+        //Recorrer la lista y hacer grupos de reglas que comparten el antecedente
+        //¿¿es necesario hacer grupos?? de momento no
+
+        ComparaGruposOrdenandoElementos compG = new ComparaGruposOrdenandoElementos();
+
+        //Guardamos la regla actual
+        Regla r_nueva = null;
+        GrupoElementos nuevo = null; //Nuevo es el antecedente comparte todo el grupo
+        int iguales = 0;
+
+        D.d("Reglas ordenadas por antecedente y consecuente");
+        for (Regla r : reglas) {
+            D.d(r.toString());
+            //Extraemos el grupo de elementos del antecedente
+            GrupoElementos e = r.getAntecedente();
+            if (nuevo == null) {//Soy el primero, me asigno el turno
+                nuevo = e;
+                r_nueva = r;
+                iguales = 0;
+            } else {
+                if (compG.compare(e, nuevo) == 0) {//Si soy igual que el nuevo
+                    //Si mi consecuente no incluye el consecuente del nuevo, entonces me sumo
+                    //if(!nuevo.contenido_en(e)){
+                    respuesta.add(r);
+                    iguales++;
+                    //}else{ //Si el consecuente del que estamos analizando contiene al del modelo
+
+                    //}
+                } else {
+                    //Si soy diferente del nuevo es que se han terminado sus iguales y comienzan los míos
+                    if (iguales > 0) {
+                        respuesta.add(r_nueva);
+                    } //Lo apunto si había más de uno
+                    nuevo = e; //Me asigno el turno
+                    r_nueva = r;
+                    iguales = 0; //Reinicio el contador de iguales
+                }
+            }
+        }
+
+        //Si había iguales en el último caso añadimos también el último
+        if (iguales > 0) {
+            respuesta.add(r_nueva);
+        }
+
+        return respuesta;
+    }
+
+    /**
+     * Añade a la historia la evaluación de los casos ambiguos
+     *
+     * @param historia
+     * @param casos_ambiguos
+     * @return
+     */
+    private boolean[][] extender_historia(boolean[][] la_historia, ArrayList<Regla> casos_ambiguos, int t_ventana) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.